### DISCUSSION TITLE
Can multiple clients play different games?

### DISCUSSION THREAD

**Discussion by Super-Beast** [LizardByte/community#262] [Sunshine]
I tried asking this earlier but maybe I wrote it in a confusing way... As the title says, can multiple clients play their own seperate games without interfering with eachother?

**Comment by KuleRucket** [comment]
Sunshine reflects what the GPU is producing for a screen connected to the host. I think that the only way you could get this to work would but to use Virtual Machines with dedicated GPUs.

**Reply by Super-Beast** [reply]
I don't know if this is possible, but is there a way to simulate multiple monitors and attach a client to each one?

**Comment by ReenigneArcher** [comment]
No. Check out Games on Whales/Wolf.

**Comment by ildevblu** [comment]
It is possible.

I'm using ASTER to create multiple separated instances that share the same computer resources.

Two users created on the same machine can run concurrently. That's simply amazing and I love it :D
I can play for example The Witcher 3 while my girlfriend is playing Hogwarts Legacy on the TV downstairs using only a single pc.

The hard part is to tell GameStream or Sunshine what user instance it should stream..
Installing Parsec in the second user and using it as a streaming service Seems to work pretty well but it's not perfect and I got some problems on a single damn xbox gamepad button.

Steam link works too but if the first user has steam opened too then sometimes it becomes confused.

I'm trying to find a solution to tell sunshine what user should be shown but at the moment I'm a bit stuck.

**Reply by ReenigneArcher** [reply]
Probably much easier and cheaper to create virtual machines with gpu passthrough.

**Reply by joydashy** [reply]
How so cheaper? That would require buying multiple gaming-level GPU's.

**Reply by ReenigneArcher** [reply]
Spending hundreds of hours developing the code to support this niche use case... that's far more expensive than a second gpu.

**Reply by joydashy** [reply]
Ah yes of course, sorry I thought I was misunderstanding the technical concept.

**Reply by ildevblu** [reply]
Actually, you don't need multiple GPUs with GPU passthrough. You can have multiple instances with a single shared GPU and allocate a percentage of its power to each VM. Thanks to ReengineArcher's suggestion, last year I started researching and found a powerful setup that allows playing games from a VM with full GPU acceleration while using the same GPU to play another game on the main system. 
Obviously, the more powerful the GPU, the better the games run

**Reply by joydashy** [reply]
You're talking about Aster or something else entirely?

**Reply by ildevblu** [reply]
a way cleaner way, by using Hyper-V to create a virtual machine and by configuring it to use the system gpu. Then installed sunshine on the vm and moonlight on the client system. The hard part is the configuration of the vm that has to share the same gpu drivers and should have the same updated Windows version

**Comment by willow-sprite** [comment]
adding another solution/workaround

2 running instances of sunshine with different conf files.
+ making a nucleus coop handler to separate the inputs for the programs you want to launch.

I would strongly recommend against using aster, as it is closed source and originates in russia.

[a comment by an automated account is not shown]

**Comment by Markus328** [comment]
You can always run a headless wayland compositor in a different user (or not). After running the compositor, start inside the sunshine service with capture = "wlr" to be able to start a full diferent session. However, note that this would only work with wlroots wayland compositors (this method at least) and the devices from sunshine will mess the other user session (the real one, not headless). An approach to this is to use games-on-whales/wolf or create a container, use [vuinputd](https://github.com/joleuger/vuinputd) and start a headless wlroots wayland compositor inside. I'm actually doing this, and it's working well. Aside I doubt is impossible to use KMS capture that would require both root and a real display
